bodo/pandas/groupby.py

```diff
@@ -83,6 +84,36 @@ def __getattribute__(self, name: str, /) -> Any:
                 gb = gb[self._selection]
             return object.__getattribute__(gb, name)
 
+    @check_args_fallback(supported="func")
+    def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs):
+        return _groupby_agg_plan(self, func, *args, **kwargs)
+
+    agg = aggregate
+
+    def _normalize_agg_func(self, func, kwargs):
+        # list of (input column name, function) pairs
+        normalized_func: list[tuple[str, str]] = []
+
+        if func is None and kwargs:
+            normalized_func = [
+                (col, _get_aggfunc_str(func_)) for col, func_ in kwargs.values()
+            ]
+        elif is_dict_like(func):
+            normalized_func = [
+                (col, _get_aggfunc_str(func_)) for col, func_ in func.items()
+            ]
+        elif is_list_like(func):
+            normalized_func = [
+                (col, _get_aggfunc_str(func_))
+                for col in self._selection
+                for func_ in func
```

Comment:
Just want to make sure this part is tested.  Not sure I know the semantics well enough to say this is wrong but just want to make sure this shouldn't be "for col, func_ in zip(self._selection, func)".